pkg/services/sqlstore/searchstore/builder.go

```diff
@@ -25,13 +25,13 @@ func (b *Builder) ToSql(limit, page int64) (string, []interface{}) {
 	b.sql.WriteString("( ")
 	b.applyFilters()
 
-	b.sql.WriteString(b.Dialect.LimitOffset(limit, (page-1)*limit) + `) as ids
-		INNER JOIN dashboard on ids.id = dashboard.id
+	b.sql.WriteString(b.Dialect.LimitOffset(limit, (page-1)*limit) + `) AS ids
+		INNER JOIN dashboard ON ids.id = dashboard.id
 	`)
 
 	b.sql.WriteString(`
-		LEFT OUTER JOIN dashboard folder on folder.id = dashboard.folder_id
-		LEFT OUTER JOIN dashboard_tag on dashboard.id = dashboard_tag.dashboard_id`)
+		LEFT OUTER JOIN dashboard AS folder ON folder.id = dashboard.folder_id
```

Comment:
Does this one make sense, as `dashboard` is present in the query as both "folder" and "dashboard"? I'm confused at least.

Reply:
I checked this with @xlson, when folders were introduced they were introduced in the dashboard table, so we're doing a left join with another "dashboard" (but without a dashboard payload) that happens to be the folder.